app/models/prison.rb

```diff
@@ -1,52 +1,84 @@
 class Prison
-  DEFAULT_LEAD_DAYS = 3.freeze
-  DEFAULT_BOOKING_WINDOW = 28.freeze
+  include Virtus.model
+
+  DEFAULT_ADULT_AGE =       18
+  DEFAULT_BOOKING_WINDOW =  28
+  DEFAULT_LEAD_DAYS =       3
+
+  attribute :address
+  attribute :adult_age, Integer,
+    default: DEFAULT_ADULT_AGE,
+    lazy: true
+  attribute :booking_window, Integer,
+    default: DEFAULT_BOOKING_WINDOW,
+    lazy: true
+  attribute :canned_responses, Boolean
+  attribute :email, String
+  attribute :enabled, Boolean
+  attribute :finder_slug, String
+  attribute :lead_days, Integer,
+    default: DEFAULT_LEAD_DAYS,
+    lazy: true
+  attribute :name, String
+  attribute :nomis_id, String
+  attribute :phone, String
+  attribute :reason
+  attribute :slot_anomalies
+  attribute :slots
+  attribute :unbookable, Boolean
+  attribute :works_weekends, Boolean,
+    default: false,
+    lazy: true
 
   class PrisonNotFound < StandardError; end
 
-  def self.find(prison_name)
-    prison_hash = Rails.configuration.prison_data[prison_name]
-    if prison_hash.nil?
-      raise PrisonNotFound, "Can't find prison #{prison_name.inspect}"
-    end
-    new(prison_name, prison_hash)
+  def self.all
+    Rails.configuration.prisons
   end
 
-  def initialize(name, prison_hash)
-    @name = name
-    @prison_hash = prison_hash.with_indifferent_access
+  def self.create(opts = {})
+    new(opts).tap { |p| Rails.configuration.prisons << p }
   end
 
-  attr_reader :name
+  def self.enabled
+    all.select(&:enabled)
+  end
 
-  delegate :fetch, to: :@prison_hash
+  def self.find(name)
+    all.detect { |p| p.name == name }.try(:dup)
+  end
 
-  def unbookable_dates
-    (fetch(:unbookable) { Array.new }).to_set
+  def self.find_by_nomis_id(nomis_id)
+    all.detect { |p| p.nomis_id == nomis_id }.try(:dup)
```

Comment:
Now that we've discovered that NOMIS IDs aren't unique per prison (or per our model of a "prison", anyway), is this method ever safe to use?